src/validate-sorted-params.ts

```diff
@@ -0,0 +1,41 @@
+import {toResult, Result, joinErrors} from './util';
+import * as S from './soy-parser';
+import * as T from 'babel-types';
+import visit from './soy-traverse';
+
+export interface paramObj extends Object {
```

Comment:
Types should be capitalized, but since this really only will ever be used here, and there are only two attributes, lets use a tuple.

So, something like this:
```typescript
const calls: Array<[string, Array<string>]> = [];

/* ... */

calls.push([name, paramNames]);

/* ... */

calls.map(([name, paramNames]) => ... );
```